dspace-api/src/test/java/org/dspace/core/UtilsTest.java

```diff
@@ -22,6 +22,27 @@
  */
 public class UtilsTest extends AbstractUnitTest {
 
+    /**
+     * Test of getBaseUrl method, of class Utils
+     */
+    @Test
+    public void testGetBaseUrl() {
+        assertEquals("Test remove /server", "http://dspace.org",
+                     Utils.getBaseUrl("http://dspace.org/server"));
+
+        assertEquals("Test remove /server/api/core/items", "https://dspace.org",
+                     Utils.getBaseUrl("https://dspace.org/server/api/core/items"));
+
+        assertEquals("Test remove trailing slash", "https://dspace.org",
+                     Utils.getBaseUrl("https://dspace.org/"));
+
+        assertEquals("Test keep url", "https://demo.dspace.org",
+                     Utils.getBaseUrl("https://demo.dspace.org"));
```

Comment:
@atarix83  What would happen if the Base URL includes a port number?  While it's less frequently seen, I have seen sites that run their DSpace on port 8080, for example.  It looks to me like `Utils.getBaseUrl()` would fail if the port is not either 80 (HTTP) or 443 (HTTPS)

Reply:
added compatibility with url that contains port